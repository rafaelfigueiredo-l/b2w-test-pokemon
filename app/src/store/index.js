import { createStore } from "redux";

const INITIAL_STATE = {
  theme: {},
  pokemon: [],
};
function course(state = INITIAL_STATE, action) {
  switch (action.type) {
    case "SET_THEME":
      return {
        ...state,
        theme: action.payload,
      };

    case "SET_POKEMON":
      return {
        ...state,
        pokemon: action.payload,
      };

    default:
      return state;
  }
}

const store = createStore(
  course,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default store;
