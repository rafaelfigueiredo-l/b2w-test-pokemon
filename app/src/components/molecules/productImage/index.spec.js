import { Provider } from "react-redux";
import store from "../../../store";
import ProductImage from "./";

describe("<ProductImage />", () => {
  const component = (
    <Provider store={store}>
      <ProductImage />
    </Provider>
  );

  const mounted = mount(component);

  it("should mount correctly ProductImage component", () => {
    expect(mounted.find(".productImage")).toHaveLength(2);
    expect(mounted).toMatchSnapshot();
  });

  it("should unmount correctly ProductImage component", () => {
    expect(mounted.unmount()).toEqual({});
  });
});
