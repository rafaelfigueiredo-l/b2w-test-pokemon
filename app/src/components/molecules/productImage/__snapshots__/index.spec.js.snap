// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductImage /> should mount correctly ProductImage component 1`] = `
<styled.div
  className="productImage"
>
  <div
    className="sc-AxiKw hIDtSO productImage"
  >
    <styled.img
      alt="Loader"
      className="loader"
      src="loader.gif"
    >
      <img
        alt="Loader"
        className="sc-AxirZ iDbaau loader"
        src="loader.gif"
      />
    </styled.img>
  </div>
</styled.div>
`;
